generator/generator.py

```diff
@@ -6,17 +6,21 @@
 import json
 import hashlib
 import shutil
+import os
 from pathlib import Path
 
 DOWNLOAD_URL = "https://downloads.arduino.cc/arduino-fwuploader"
-FQBNS = {
-    "mkr1000": "arduino:samd:mkr1000",
-    "mkrwifi1010": "arduino:samd:mkrwifi1010",
-    "nano_33_iot": "arduino:samd:nano_33_iot",
-    "mkrvidor4000": "arduino:samd:mkrvidor4000",
-    "uno2018": "arduino:megaavr:uno2018",
-    "nanorp2040connect": "arduino:mbed_nano:nanorp2040connect",
-}
+
+
+# handle firmware name
+def get_firmware_name(fqbn):
+    if fqbn == "arduino:megaavr:uno2018":
+        return "NINA_W102-arduino.megaavr.uno2018.bin"
+    elif fqbn == "arduino:mbed_nano:nanorp2040connect":
+        return "NINA_W102-arduino.mbed_nano.nanorp2040connect.bin"
+    elif fqbn == "arduino:samd:mkr1000":
+        return "m2m_aio_3a0-arduino.samd.mkr1000.bin"
+    return "NINA_W102.bin"
```

Comment:
You should be able to make this generic I think, I was thinking something like this.

```
def get_firmware_file(module, fqbn):
    fqbn = '.'.join(fqbn.split(':'))
    file = Path(f'{module}-{fqbn}.bin')
    if file.exists():
        return file
    return Path(f'{module}.bin')
```

Reply:
Obviously that would mean that the `moduleName` key in the `boards.json` must match the firmware files name.